flowauth/backend/tests/conftest.py

```diff
@@ -141,93 +134,132 @@ def test_two_factor_auth_user(app, get_two_factor_code):
-
-        test_group = Group.query.filter(Group.id == test_group.id).first()
-        # Each user is also a group
-        for user in User.query.all():
-            test_group.members.append(user)
-        db.session.add(test_group)
-
         # Add some servers
         dummy_server_a = Server(
             name="DUMMY_SERVER_A",
-            longest_token_life=2,
+            longest_token_life_minutes=24 * 60 * 2,
             latest_token_expiry=datetime.datetime.now().date()
             + datetime.timedelta(days=365),
         )
         dummy_server_b = Server(
             name="DUMMY_SERVER_B",
-            longest_token_life=2,
+            longest_token_life_minutes=24 * 60 * 2,
             latest_token_expiry=datetime.datetime.now().date()
-            + datetime.timedelta(days=700),
+            + datetime.timedelta(days=365),
         )
         db.session.add(dummy_server_a)
         db.session.add(dummy_server_b)
+        db.session.commit()
+        return (dummy_server_a, dummy_server_b)
+
+
+@pytest.fixture  # (scope="session")
+def test_scopes(app, test_servers):
+    with app.app_context():
+        dummy_server_a, dummy_server_b = test_servers
+        scopes = [
+            read_scope_a := Scope(
+                name="get_result",
+                server=dummy_server_a,
+            ),
+            read_scope_b := Scope(name="get_result", server=dummy_server_b),
+            run_scope := Scope(name="run", server=dummy_server_a),
+            dummy_query_scope := Scope(
+                name="dummy_agg_unit:dummy_query:dummy_query", server=dummy_server_a
+            ),
+        ]
+        db.session.add_all(scopes)
+        db.session.commit()
+        return scopes
+
+
+@pytest.fixture  # (scope="session")
+@freeze_time("2020-12-31")
+def test_roles(app, test_scopes, test_servers):
+    with app.app_context():
+        read_a, read_b, run, dummy_query = test_scopes
+        server_a, server_b = test_servers
+        runner = Role(
+            name="runner",
+            scopes=[run, read_a, dummy_query],
+            server=server_a,
+            longest_token_life_minutes=24 * 60 * 2,
+            latest_token_expiry=datetime.datetime.now() + datetime.timedelta(minutes=5),
+        )
+        reader = Role(
+            name="reader",
+            scopes=[read_a],
+            server=server_a,
+            longest_token_life_minutes=24 * 60 * 2,
+            latest_token_expiry=datetime.datetime.now() + datetime.timedelta(minutes=5),
+        )
+        reader_b = Role(
+            name="reader_b",
+            scopes=[read_b],
+            server=server_b,
+            longest_token_life_minutes=24 * 60 * 2,
+            latest_token_expiry=datetime.datetime.now() + datetime.timedelta(minutes=5),
+        )
+        db.session.add(runner)
+        db.session.add(reader)
+        db.session.add(reader_b)
+        db.session.commit()
+        return runner, reader, reader_b
+
+
+@pytest.fixture  # (scope="session")
```

Comment:
```suggestion
@pytest.fixture
```

There are some other instances of commenting out in preference to deleting - better to delete, git remembers it was there.